### ISSUE TITLE
Legacy extension

### ISSUE THREAD

**Issue by jonathanKingston** [firebug/websocket-monitor#75]
As this extension is currently using SDK code that will be blocked in 57, will this be workable after 57?

In #69 it was discussed if this would merge with devtools itself however I don't know if that will happen.

**Comment by CamJN** [comment]
Any word on this?

**Comment by eliihen** [comment]
To make the conversion to the WebExtension API happen, we need to rewrite the files in the `lib` directory. Not sure if we need to touch anything in the `data` directory, but a quick search seems like it is unaffected. 

```
$ grep -rP 'require\("(?!react|\.|redux|reducers|firebug|reps)' lib/ data/
lib/wsm-actor.js:const { Cc, Ci, Cu } = require("chrome");
lib/wsm-actor.js:const Events = require("sdk/event/core");
lib/mqtt-parser.js:}).call(this,require("buffer").Buffer)
lib/mqtt-parser.js:}).call(this,require("buffer").Buffer)
lib/mqtt-parser.js:}).call(this,require("buffer").Buffer)
lib/mqtt-parser.js:}).call(this,require("buffer").Buffer)
lib/mqtt-parser.js:}).call(this,require("buffer").Buffer)
lib/mqtt-parser.js:}).call(this,require("buffer").Buffer)
lib/wsm-toolbox-overlay.js:const options = require("@loader/options");
lib/wsm-toolbox-overlay.js:const { Cu, Ci } = require("chrome");
lib/wsm-toolbox-overlay.js:const { Class } = require("sdk/core/heritage");
lib/wsm-netmonitor-overlay.js:const { Cu, Ci } = require("chrome");
lib/wsm-netmonitor-overlay.js:const { Class } = require("sdk/core/heritage");
lib/wsm-netmonitor-overlay.js:const { loadSheet, removeSheet } = require("sdk/stylesheet/utils");
lib/wsm-panel.js:const options = require("@loader/options");
lib/wsm-panel.js:const { Cc, Cu, Ci } = require("chrome");
lib/wsm-panel.js:const { Class } = require("sdk/core/heritage");
lib/wsm-panel.js:const { Tool } = require("dev/toolbox");
lib/wsm-panel.js:const { defer, resolve, all } = require("sdk/core/promise");
lib/wsm-panel.js:const { prefs }  = require("sdk/simple-prefs");
lib/index.js:let { main, Loader, override } = require("toolkit/loader");
lib/index.js:let options = require("@loader/options");
lib/index.js:let defaultGlobals = override(require("sdk/system/globals"), {
lib/index.js:   "sdk/addon/window": require("sdk/addon/window"),
```

- [See: manifest devtools_page](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/manifest.json/devtools_page)
- [See: Extending the developer tools](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/Extending_the_developer_tools)
- [See: Comparison with the Add-on SDK](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/Comparison_with_the_Add-on_SDK)

**Comment by eliihen** [comment]
Whelp, firebug.sdk, **the** core dependency of the view layer doesn't work with WebExtensions. Nor can I really see how it could in any reasonable amount of time.

@janodvarko Guess that means websocket-monitor requires a complete ground-up rewrite. Agree?

**Comment by eliihen** [comment]
I inquired a bit about this, and it does require an extension to the WebExtension API in order to provide the WebSocket Monitor as a WebExtension: https://bugzilla.mozilla.org/show_bug.cgi?id=1373639#c6

Honza played around with this a few months ago, the repo is here: https://github.com/janodvarko/webext-websocket-monitor

**Comment by rhessinger** [comment]
Any update on this?  The new version of Firefox Developer Edition 57+ will not allow this extension.

**Comment by eliihen** [comment]
Not that I know - @janodvarko has not yet reached out

**Comment by janodvarko** [comment]
> @janodvarko Guess that means websocket-monitor requires a complete ground-up rewrite. Agree?

Yes, complete rewrite needed. I already started with a prototyping here:
https://github.com/janodvarko/webext-websocket-monitor
To see how to get the backend WS API.

I am planning to work on this more in Q4 2017

Honza

**Comment by eliihen** [comment]
@janodvarko I would love to help out with the rewrite! I tried to get the prototype running, but was unable to make it work, see https://github.com/janodvarko/webext-websocket-monitor/issues/2

**Comment by janodvarko** [comment]
Awesome, I'll get back to this ASAP and I'll keep you in the loop!

Honza

**Comment by Croydon** [comment]
Users are asking about his now since we are past the FF57 release. Any updates?

**Comment by janodvarko** [comment]
Sorry, no progress so far, might move to Q1 2018

But, the extension has priority and is being on top of our todo list!

Honza

**Comment by XaviTorello** [comment]
@janodvarko is there any planning to perform the refactor? I would contribute if possible

**Comment by janodvarko** [comment]
> @janodvarko is there any planning to perform the refactor? I would contribute if possible

Excellent, I'll ping you as soon as the work starts on this (probably Q1/2018)

Honza

**Comment by XaviTorello** [comment]
Perfect, keep in touch about this

**Comment by TriMoon** [comment]
> @janodvarko 
Sorry, no progress so far, might move to Q1 2018

Any update on the efforts is welcome 😜

**Comment by luislobo** [comment]
Hi, Is there anything we can do to contribute? Q1 is almost gone.. :alarm_clock:

**Comment by janodvarko** [comment]
Hi,
yes sorry, this needs more time. But, I am already working on documentation (list of requirements) and I'll make sure this thread will be updated when the implementation starts!

Honza

**Comment by luislobo** [comment]
@janodvarko thank you!

**Comment by leus** [comment]
Are there any updates on this matter?

**Comment by janodvarko** [comment]
Update:
According to the latest DevTools team planning - work on the rewrite should happen later this year (end of Q3)
Honza
